compiler/src/iree/compiler/Codegen/SPIRV/KernelConfig.cpp

```diff
@@ -172,22 +172,64 @@ LogicalResult setMatmulOpConfig(linalg::LinalgOp op, int64_t subgroupSize,
                                 std::array<int64_t, 2> bestWorkgroupSizeXY,
                                 std::array<int64_t, 3> bestThreadTileSizeMNK,
                                 bool useWorkgroupMemory) {
-  auto lhsType = op.inputs()[0].getType().cast<ShapedType>();
+  OpOperand *lhs = op.getInputOperand(0);
+  OpOperand *rhs = op.getInputOperand(1);
+
+  auto lhsType = lhs->get().getType().cast<ShapedType>();
+  auto rhsType = rhs->get().getType().cast<ShapedType>();
   auto elementBits = lhsType.getElementType().getIntOrFloatBitWidth();
   if (elementBits != 16 && elementBits != 32) return success();
 
-  ArrayRef<int64_t> lhsShape =
-      op.getInputOperand(0)->get().getType().cast<ShapedType>().getShape();
-  ArrayRef<int64_t> rhsShape =
-      op.getInputOperand(1)->get().getType().cast<ShapedType>().getShape();
+  ArrayRef<int64_t> lhsShape = lhsType.getShape();
+  ArrayRef<int64_t> rhsShape = rhsType.getShape();
   if (llvm::any_of(lhsShape, ShapedType::isDynamic)) return success();
   if (llvm::any_of(rhsShape, ShapedType::isDynamic)) return success();
 
-  bool isBM = isa<linalg::BatchMatmulOp>(op);
+  assert(llvm::is_contained({2u, 3u}, op.getNumParallelLoops()));
```

Comment:
I think there is a recognizer somewhere that might do this for you . @ThomasRaoux ?

Reply:
unfortunately I don't think so. Are you talking about `isaContractionOpInterface`? This is all we have but it is a bit more relaxed that what we want right now.

Reply:
Given that matchers of Linalg ops is a thing that has been stated as something good to have, maybe this can just be added upstream into family of matchers that exist for Linalg